File: frontend/src/types/material.ts
// Mirrors backend/app/schemas/material.py so the API contract only has one
// place it can drift: this file.

export type MaterialKind = "resume" | "cover_letter" | "portfolio" | "other";

export const MATERIAL_KIND_LABELS: Record<MaterialKind, string> = {
  resume: "Résumé",
  cover_letter: "Cover letter",
  portfolio: "Portfolio",
  other: "Other",
};

export const MATERIAL_KIND_ORDER: MaterialKind[] = ["resume", "cover_letter", "portfolio", "other"];

export interface MaterialPublic {
  id: string;
  kind: MaterialKind;
  name: string;
  asset_url: string;
  mime_type: string;
  bytes: number;
  created_at: string;
  updated_at: string;
}

export interface MaterialListResponse {
  items: MaterialPublic[];
}

File: backend/app/schemas/material.py
"""Request/response contracts for the materials (résumé/upload) endpoints."""

from datetime import datetime
from typing import List

from pydantic import BaseModel

from app.models.material import MaterialKind


class MaterialPublic(BaseModel):
    id: str
    kind: MaterialKind
    name: str
    asset_url: str
    mime_type: str
    bytes: int
    created_at: datetime
    updated_at: datetime


class MaterialListResponse(BaseModel):
    items: List[MaterialPublic]
